core/src/main/scala/kafka/tools/GetOffsetShell.scala

```diff
@@ -135,7 +135,11 @@ object GetOffsetShell {
       val partitionOffsets = partitionInfos.flatMap { tp =>
         try {
           val partitionInfo = listOffsetsResult.partitionResult(tp).get
-          Some((tp, partitionInfo.offset))
+          if (partitionInfo.offset != ListOffsetsResponse.UNKNOWN_OFFSET) {
+            Some((tp, partitionInfo.offset))
+          } else {
+            None
```

Comment:
Could we output something to user that like the consumer group command did: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala#L680-L681

ex: "Warn: the offset for Partition $tp is unknown." WDYT?

Reply:
Good suggestion, we can print a reminder here.

Reply:
Using a warning does not seem right here as it is expected to have no offsets in certain conditions. In the previous implementation, they were just ignored. Why not doing the same to be consistent? If we believe that this is not good enough, perhaps printing the -1 (or NONE) to indicate there is no offset for a partition might be better.

Reply:
Yeah, printing the -1 or printing a message both are ok, but we can't always ensure -1 represents no offsets in the future, printing -1 is a better way forward. then we should improve the description of the --time arg.

Reply:
>  In the previous implementation, they were just ignored. Why not doing the same to be consistent?

One thing to clarify: @dengziming , I think in the previous implementation is returning `empty` offset, not `-1`, is that right?

Reply:
Yes, nothing is printed if no offset, for example, when the topic record version is 0.8.0 which has no timestamp.

Reply:
In that case, if we want to be consistent with previous behavior, should we return `empty`, instead of `-1`? WDYT?

Reply:
The previous implement is controlled by KafkaConsumer and we have no ways to change it,  however, now we can improve the output since AdminClient give us this information. 
Here we can either print an info message or print -1 to show this information, printing -1 is more reasonable since it's expected to have no offsets in certain conditions as David commented.
Do you think this makes sense?

Reply:
Make sense. Thanks.